app/models/user.rb

```diff
@@ -1963,6 +1969,47 @@ def set_default_sidebar_section_links
     SidebarSectionLink.insert_all(records) if records.present?
   end
 
+  def set_default_sidebar_section_links
+    return if !SiteSetting.enable_experimental_sidebar_hamburger
+    return if staged? || bot?
+
+    records = []
+
+    if SiteSetting.default_sidebar_categories.present?
+      default_category_ids = SiteSetting.default_sidebar_categories.split("|")
+
+      # Filters out categories that user does not have access to or do not exist anymore
+      category_ids = Category.secured(self.guardian).where(id: default_category_ids).pluck(:id)
+      existing_category_ids = SidebarSectionLink.where(user: self, linkable_type: 'Category').pluck(:linkable_id)
+
+      # Only include secured categories
+      secured_category_ids = self.secure_category_ids
+
+      # Secured categories that are in the default categories list
+      categories_to_add = existing_category_ids + (category_ids & secured_category_ids)
+
+      updater = UserUpdater.new(self, self)
```

Comment:
I think it would be good to re-use the existing code we have for this in `UserUpdater`, but I'm not sure if calling it with `self, self` is the proper way? If not maybe we need to move it that logic to somewhere else that we can access from the User Model as well as the UserUpdater?

Reply:
I thought about what you said here and I agree that we should probably have a dedicated class to handle the updating of sidebar section links :+1:  I introduced `SidebarSectionLinksUpdater` in https://github.com/discourse/discourse/pull/19275 which you can use in this PR.